src/components/tree/phyloTree/change.ts

```diff
@@ -74,33 +80,42 @@ const svgSetters = {
  * the SVG elements.
  * svgSetters (see above) are used to actually modify the property on the element,
  * so the given property must also be present there!
- * @param {string} treeElem (e.g. ".tip" or ".branch")
- * @param {list} properties (e.g. ["visibiliy", "stroke-width"])
- * @return {function} used in a d3 selection, i.e. d3.selection().methods().call(X)
  */
-const createUpdateCall = (treeElem, properties) => (selection) => {
-  // First: the properties to update via d3Selection.attr call
-  if (svgSetters.attrs[treeElem]) {
-    [...properties].filter((x) => svgSetters.attrs[treeElem][x])
-      .forEach((attrName) => {
-        // console.log(`applying attr ${attrName} to ${treeElem}`)
-        selection.attr(attrName, svgSetters.attrs[treeElem][attrName]);
-      });
-  }
-  // Second: the properties to update via d3Selection.style call
-  if (svgSetters.styles[treeElem]) {
-    [...properties].filter((x) => svgSetters.styles[treeElem][x])
-      .forEach((styleName) => {
-        // console.log(`applying style ${styleName} to ${treeElem}`)
-        selection.style(styleName, svgSetters.styles[treeElem][styleName]);
-      });
-  }
-};
+function createUpdateCall(
+  treeElem: TreeElement,
+
+  /** e.g. ["visibility", "stroke-width"] */
+  properties: Set<SVGProperty>,
+):       (selection: Transition<SVGGElement, PhyloNode, SVGSVGElement, unknown>) => void {
+  return (selection: Transition<SVGGElement, PhyloNode, SVGSVGElement, unknown>) => {
```

Comment:
Wow I find TS hard to read sometimes!

Here we're typing the returned function signature twice, but (I think) we can drop the second duplicate signature.

```ts
function createUpdateCall(
  treeElem: TreeElement,
  properties: Set<SVGProperty>
): (selection: Transition<SVGGElement, PhyloNode, SVGSVGElement, unknown>) => void {
  return (selection) => {
    ...
```

Or even clearer (IMO):

```ts
type UpdateCall = (selection: Transition<SVGGElement, PhyloNode, SVGSVGElement, unknown>) => void;

function createUpdateCall(treeElem: TreeElement, properties: Set<SVGProperty>): UpdateCall {
  return (selection) => {
    ...
  }
}
```

Reply:
After rebasing onto latest changes, I updated it to: https://github.com/nextstrain/auspice/blob/ad1f2cedbde4df6974412d37fd84119208902065/src/components/tree/phyloTree/change.ts#L77-L81